lib/orbf/rules_engine/builders/zone_activity_formula_variables_builder.rb

```diff
@@ -0,0 +1,145 @@
+# frozen_string_literal: true
+
+module Orbf
+  module RulesEngine
+    class ZoneActivityFormulaVariablesBuilder
+      include VariablesBuilderSupport
+
+      def initialize(package, orgunits, period)
+        @package = package
+        @ref_orgunit = orgunits.ref_orgunit
+        @secondary_orgunits = orgunits.secondary_orgunits
+        @period = period
+      end
+
+      # Gets called by SolverFactory to
+      def to_variables
+        return [] unless package.zone_activity_rules.any?
+
+        zone_activity_formula_variables
+      end
+
+      private
+
+      attr_reader :orgunits, :package, :period, :ref_orgunit, :secondary_orgunits
+
+      # Returns an array of Orbf::RulesEngine::Variable
+      def zone_activity_formula_variables
+        org_units_count = secondary_orgunits.size.to_s
+        org_units_key = Orbf::RulesEngine::ContractVariablesBuilder::ORG_UNITS_COUNT
+
+        package.all_activities_codes.each_with_object([]) do |activity_code, array|
+          package.zone_activity_rules.each do |rule|
+            rule.formulas.each do |formula|
+              values = activity_formula_values(formula, activity_code)
```

Comment:
with the comment... may be we should name them 

```ruby
ORG_UNITS_COUNT_KEY = Orbf::RulesEngine::ContractVariablesBuilder::ORG_UNITS_COUNT

def 
    ....
      rule.formulas.each do |formula|
            expanded_expression = expand_percent_values(formula, activity_code)
            expanded_expression = expand_direct_variable_reference(expanded_expression,formula, activity_code)
            array.push(build_variable(activity_code, formula, expanded_expression))
     end

# Expand %{vars}
def expand_percent_values(formula, activity_code)
        values = activity_formula_values(formula, activity_code)
        formula.expression % values
end

# Expand direct_vars
def expand_direct_variable_reference(expression,formula, activity_code)
        org_units_count = secondary_orgunits.size.to_s
        direct_variables = all_zone_referenced_formula(formula, activity_code)
        direct_variables[ORG_UNITS_COUNT_KEY] = org_units_count
        Tokenizer.replace_token_from_expression(expression, direct_variables, {})
```

Reply:
may be I should start having a clearer vocabulary, what do you think about : 

- `expand` for `%{}` (and instead of values name them expansions)
- `substitute_references` (vs expand_direct_variable_reference) (and instead direct_variables name them substitutions )

or other possible name, at some point I was speaking of instantiation when
 -  the user provided expression `formula_code := xx * 2` 
 - is turned into the `package_code_formula_code_for_orgunit_and_period := xx_for_orgunit_period * 2 ` form 

we can rename the methods : 
   `activity_formula_values` => `activity_formula_expansions`
   `all_zone_referenced_formula` => `all_zone_referenced_formula_substitutions`